src/lib/ffi/ffi.h

```diff
@@ -259,6 +260,33 @@ BOTAN_FFI_EXPORT(2, 3) int botan_base64_decode(const char* base64_str, size_t in
 */
 typedef struct botan_rng_struct* botan_rng_t;
 
+/**
```

Comment:
These declarations are in the middle of the RNG declarations. They should all be moved, probably down to the very end of the file.